src/tensor/tensor.ts

```diff
@@ -128,20 +137,33 @@ class Tensor {
   requires_grad = false
   grad: Tensor = null
   op = 'constant'
+  op_count = 0
 
   _injest_ptr(_ptr) {
     const numel = Number(fl.elements.native(_ptr))
     // eslint-disable-next-line @typescript-eslint/ban-ts-comment
     // @ts-ignore - overload toArrayBuffer params
     this.underlying = toArrayBuffer(_ptr, 0, numel * 4, fl.genTensorDestroyer.native())
+    /* TODO: possibly use deallocatorContext? */
+    // this.underlying = toArrayBuffer(_ptr, 0, numel * 4, _ptr, fl.genTensorDestroyer.native())
   }
 
   backward(jacobian) {
     return backward(this, jacobian)
   }
 
+  cleanUp() {
+    if (this.op_count >= 50) {
```

Comment:
I wonder if this scales to different types of operations/sizes.  I'm not sure about arrayfire, but ideally we make this generic/amenable to other JITs as well.

Reply:
Added EvalTuner class as per our conversation on discord; update eval.test.ts to use EvalTuner (and tinkered w defaults to find values that outperform my naive manual hillclimb search yesterday in testing).